src/overlays/actors/ovl_En_Test5/z_en_test5.c

```diff
@@ -31,14 +28,83 @@ const ActorInit En_Test5_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void EnTest5_SetupAction(EnTest5* this, EnTest5ActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnTest5_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    EnTest5* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    WaterBox* water;
+    f32 ySurface; // Unused
+
+    // If not spawned above a water source, immediately despawn
+    if (!WaterBox_GetSurface1(globalCtx, &globalCtx->colCtx, this->actor.world.pos.x, this->actor.world.pos.z,
+                              &ySurface, &water)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    Math_Vec3s_ToVec3f(&this->minPos, &water->minPos);
+    this->xLength = (f32)water->xLength;
+    this->zLength = (f32)water->zLength;
+
+    EnTest5_SetupAction(this, EnTest5_HandleBottleAction);
+}
+
+void EnTest5_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnTest5* this = THIS;
+}
+
+void EnTest5_HandleBottleAction(EnTest5* this, GlobalContext* globalCtx) {
+    Player* player;
+    Vec3f playerPosRelativeToWater;
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.parent = NULL;
+        return;
+    }
+
+    player = GET_PLAYER(globalCtx);
+
+    if (player->unk_388 == NULL || player->unk_384 != GI_MAX) {
+        Math_Vec3f_DistXYZAndStoreDiff(&this->minPos, &player->actor.world.pos, &playerPosRelativeToWater);
+
+        // Make sure that the player is within the bounds of the water and deep enough to grab some
+        if (playerPosRelativeToWater.x >= 0.0f && playerPosRelativeToWater.x <= this->xLength &&
+            playerPosRelativeToWater.z >= 0.0f && playerPosRelativeToWater.z <= this->zLength &&
+            fabsf(playerPosRelativeToWater.y) < 100.0f && player->actor.depthInWater > 12.0f) {
+            func_800B8A1C(&this->actor, globalCtx, GI_MAX, this->actor.xzDistToPlayer,
+                          fabsf(this->actor.playerHeightRel));
+        }
+    }
+}
+
+void EnTest5_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnTest5* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    Vec3f steamPos;
+    CollisionPoly* poly; // Unused
+    UNK_TYPE pad;        // Unused
```

Comment:
```suggestion
    s32 pad;        // Unused
```
Generally just use `s32` instead of `UNK_TYPE` for pads.